### ISSUE TITLE
protobuf 3.7.0 is failing to build with closure compiler

### ISSUE THREAD

**Issue by davido** [protocolbuffers/protobuf#5931]
**What version of protobuf and what language are you using?**
Version: v3.7.0.
Language: Javascript

**What operating system?**

Linux

**What runtime / compiler are you using (e.g., python version or gcc version)**

Python 3.7.2
gcc 8.3.1

**What did you do?**
1. Upload this PR https://github.com/bazelbuild/rules_closure/pull/342 to upgrade Protobuf version to 3.7.0 in rules_closure, that is used in Gerrit Code Review
2. Run the tests with: `bazel test closure/...`

**What did you expect to see**

Tests are passing

**What did you see instead?**

Tests are failing:

```
external/com_google_protobuf_js/message.js:44: ERROR - extra require: 'jspb.BinaryConstants' is never referenced in this file
goog.require('jspb.BinaryConstants');
^
  ProTip: "JSC_EXTRA_REQUIRE_WARNING" or "extraRequire" or "legacyGoogScopeRequire" can be added to the `suppress` attribute of:
  //closure/protobuf:jspb
  Alternatively /** @suppress {legacyGoogScopeRequire} */ can be added to the source file.

1 error(s), 0 warning(s), 97.2% typed

INFO: Elapsed time: 201.249s, Critical Path: 84.90s
INFO: 1261 processes: 337 linux-sandbox, 1 local, 923 worker.
````

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

**Anything else we should know about your project / environment**

All is fine with Protobuf 3.6.1.3, before the upgrade.

**Comment by davido** [comment]
When the first problem is fixed or suppressed, the next one is showing up:

```
INFO: Analysed 1246 targets (1 packages loaded, 35 targets configured).
INFO: Found 1169 targets and 77 test targets...
ERROR: /home/davido/projects/davido_rules_closure/closure/protobuf/test/BUILD:85:1: Compiling 289 JavaScript files to closure/protobuf/test/baz_test_bin.js failed (Exit 1)
bazel-out/k8-fastbuild/bin/closure/protobuf/test/bar_proto.js:71: ERROR - variable f is undeclared
    foo: (f = msg.getFoo()) && proto.io.bazel.rules.closure.protobuf.Foo.toObject(includeInstance, f)
          ^
  ProTip: "JSC_UNDEFINED_VARIABLE" or "checkVars" or "missingSourcesWarnings" or "undefinedVars" can be added to the `suppress` attribute of:
  //closure/protobuf/test:bar_proto
  Alternatively /** @suppress {undefinedVars} */ can be added to the source file.

1 error(s), 0 warning(s)

INFO: Elapsed time: 8.083s, Critical Path: 6.37s
INFO: 31 processes: 19 linux-sandbox, 1 local, 11 worker.
FAILED: Build did NOT complete successfully
```

The generated code is here:

```javascript
/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Whether to include the JSPB
 *     instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.io.bazel.rules.closure.protobuf.Bar} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.io.bazel.rules.closure.protobuf.Bar.toObject = function(includeInstance, msg) {
  var obj = {
    foo: (f = msg.getFoo()) && proto.io.bazel.rules.closure.protobuf.Foo.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
```

**Comment by dankurka** [comment]
Fix just got merged:
https://github.com/protocolbuffers/protobuf/pull/5932

**Comment by davido** [comment]
It seems that the fix for the first problem is also available: https://github.com/protocolbuffers/protobuf/pull/5933/commits/0f1d9749f1898ec953e18300f673c6529a2894ac.

@acozzette @TeBoring Can you help with the merge?

Thanks!

**Comment by acozzette** [comment]
Sure, I just merged Dan's fixes in #5933 and #5934.

**Comment by davido** [comment]
> Sure, I just merged Dan's fixes in #5933 and #5934.

Great, thanks for your help!
